src/commands/service/create-revision.ts

```diff
@@ -122,54 +94,131 @@ export const createServiceRevisionCommandDecorator = (
-        if (typeof description !== "string") {
-          throw Error(
-            `--description must be of type 'string', ${typeof description} given.`
-          );
-        }
-        if (typeof remoteUrl !== "string") {
-          throw Error(
-            `--remote-url must be of type 'string', ${typeof remoteUrl} given.`
-          );
-        }
-        if (typeof sourceBranch !== "string") {
-          throw Error(
-            `--source-branch must be of type 'string', ${typeof sourceBranch} given.`
-          );
-        }
-        if (typeof personalAccessToken !== "string") {
-          throw Error(
-            `--personal-access-token must be of type 'string', ${typeof personalAccessToken} given.`
-          );
-        }
-        if (typeof orgName !== "string") {
-          throw Error(
-            `--org-name must be of type 'string', ${typeof orgName} given.`
-          );
-        }
-
-        ////////////////////////////////////////////////////////////////////////
-        // Main
-        ////////////////////////////////////////////////////////////////////////
-        // Make a PR against all default rings
-        for (const ring of defaultRings) {
-          if (typeof title !== "string") {
-            title = `[SPK] ${sourceBranch} => ${ring}`;
-            logger.info(`--title not set, defaulting to: '${title}'`);
-          }
-          await createPullRequest(title, sourceBranch, ring, {
-            description,
-            orgName,
-            originPushUrl: remoteUrl,
-            personalAccessToken
-          });
-        }
+        await makePullRequest(
+          defaultRings,
+          title,
+          sourceBranch,
+          description,
+          orgName,
+          remoteUrl,
+          personalAccessToken
+        );
       } catch (err) {
         logger.error(err);
         process.exitCode = 1;
       }
     });
 };
+
+/**
+ * Gets the default rings
+ * @param targetBranch Target branch/ring to create a PR against
+ * @param bedrockConfig The bedrock configuration file
+ */
+export const getDefaultRings = (
+  targetBranch: any,
+  bedrockConfig: IBedrockFile
+): string[] => {
+  const defaultRings: string[] = targetBranch
+    ? [targetBranch]
+    : Object.entries(bedrockConfig.rings || {})
+        .map(([branch, config]) => ({ branch, ...config }))
+        .filter(ring => !!ring.isDefault)
+        .map(ring => ring.branch);
+  if (defaultRings.length === 0) {
+    throw Error(
+      `Default branches/rings must either be specified in ${join(
+        __dirname,
+        "bedrock.yaml"
+      )} or provided via --target-branch`
+    );
+  }
+  logger.info(
+    `Creating pull request against branches: ${defaultRings.join(", ")}`
+  );
+  return defaultRings;
+};
+
+/**
+ * Gets the source branch or parses git for the source branch
+ * @param sourceBranch The source branch
+ */
+export const getSourceBranch = async (sourceBranch: any): Promise<string> => {
```

Comment:
`sourceBranch: any` should be `sourceBranch: string | undefined`
we are avoiding `any` :-)